diff --git a/src/dlio/odom.cc b/src/dlio/odom.cc
index 7f72c4f..e7be7d3 100755
--- a/src/dlio/odom.cc
+++ b/src/dlio/odom.cc
@@ -364,2 +364,3 @@
   this->publishCloud(published_cloud, T_cloud);
+  if (this->path_pub->get_subscription_count() == 0) { return; }
 
@@ -507,2 +508,6 @@
 
+  this->scan_header_stamp = pc->header.stamp;
+  this->original_scan = original_scan_;
+  if (original_scan_->empty()) { return; }
+
   // automatically detect sensor type
@@ -529,5 +534,2 @@
 
-  this->scan_header_stamp = pc->header.stamp;
-  this->original_scan = original_scan_;
-
 }
@@ -776,2 +778,7 @@
 
+  if (this->original_scan->empty()) {
+    this->main_loop_running = false;
+    return;
+  }
+
   // Preprocess points
@@ -1425,3 +1432,3 @@
 
-  for (int i = 0; i <= this->original_scan->points.size(); i++) {
+  for (int i = 0; i < this->original_scan->points.size(); i++) {
     float d = std::sqrt(pow(this->original_scan->points[i].x, 2) +
