diff --git a/src/dlio/odom.cc b/src/dlio/odom.cc
index 4e9f190..7f72c4f 100755
--- a/src/dlio/odom.cc
+++ b/src/dlio/odom.cc
@@ -640,7 +640,7 @@ void dlio::OdomNode::deskewPointcloud() {
                         boost::range::index_value<PointType&, long> p2)
       { return p1.value().timestamp != p2.value().timestamp; };
     extract_point_time = [&sweep_ref_time](boost::range::index_value<PointType&, long> pt)
-      { return pt.value().timestamp * 1e-9f; };
+      { return pt.value().timestamp * 1e-9; };
   }
 
   // copy points into deskewed_scan_ in order of timestamp
@@ -965,6 +965,7 @@ void dlio::OdomNode::callbackImu(const sensor_msgs::msg::Imu::SharedPtr imu_raw)
                                              << to_string_with_precision(this->state.b.gyro[2], 8) << std::endl;
       }
 
+      this->prev_imu_stamp = imu_stamp_secs;
       this->imu_calibrated = true;
 
     }
